In [5]:
import pandas as np
import seaborn as sns


In [6]:
sns.get_dataset_names()

['anagrams',
 'anscombe',
 'attention',
 'brain_networks',
 'car_crashes',
 'diamonds',
 'dots',
 'dowjones',
 'exercise',
 'flights',
 'fmri',
 'geyser',
 'glue',
 'healthexp',
 'iris',
 'mpg',
 'penguins',
 'planets',
 'seaice',
 'taxis',
 'tips',
 'titanic']

In [7]:
df = sns.load_dataset('titanic')

In [8]:
df.head()

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   
3         1       1  female  35.0      1      0  53.1000        S  First   
4         0       3    male  35.0      0      0   8.0500        S  Third   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True  
3  woman       False    C  Southampton   yes  False  
4    man        True  NaN  Southampton    no   True

In [9]:
df.head(2)

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False

In [10]:
df.tail(2)

survived  pclass   sex   age  sibsp  parch   fare embarked  class  who  \
889         1       1  male  26.0      0      0  30.00        C  First  man   
890         0       3  male  32.0      0      0   7.75        Q  Third  man   

     adult_male deck embark_town alive  alone  
889        True    C   Cherbourg   yes   True  
890        True  NaN  Queenstown    no   True

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype   
---  ------       --------------  -----   
 0   survived     891 non-null    int64   
 1   pclass       891 non-null    int64   
 2   sex          891 non-null    object  
 3   age          714 non-null    float64 
 4   sibsp        891 non-null    int64   
 5   parch        891 non-null    int64   
 6   fare         891 non-null    float64 
 7   embarked     889 non-null    object  
 8   class        891 non-null    category
 9   who          891 non-null    object  
 10  adult_male   891 non-null    bool    
 11  deck         203 non-null    category
 12  embark_town  889 non-null    object  
 13  alive        891 non-null    object  
 14  alone        891 non-null    bool    
dtypes: bool(2), category(2), float64(2), int64(4), object(5)
memory usage: 80.7+ KB


# Here in the above dataset ,there are 891 entries i.e row count is also :891 where in the above data we find 15 number of columns.

In [12]:
df.count()

survived       891
pclass         891
sex            891
age            714
sibsp          891
parch          891
fare           891
embarked       889
class          891
who            891
adult_male     891
deck           203
embark_town    889
alive          891
alone          891
dtype: int64

In [13]:
df.isnull().sum()

survived         0
pclass           0
sex              0
age            177
sibsp            0
parch            0
fare             0
embarked         2
class            0
who              0
adult_male       0
deck           688
embark_town      2
alive            0
alone            0
dtype: int64

# In the above table ,we can find the missing value where we can see Column(Age) contains 177 numbers of missing values,Similarly in column (deck) we can see 688 missing value whereas in column (embark_town) we get 2 missing values.

In [14]:
df.dtypes

survived          int64
pclass            int64
sex              object
age             float64
sibsp             int64
parch             int64
fare            float64
embarked         object
class          category
who              object
adult_male         bool
deck           category
embark_town      object
alive            object
alone              bool
dtype: object

In [15]:
df.age.dtypes

dtype('float64')

In [16]:
df.shape

(891, 15)

In [17]:
rows, columns = df.shape
print(f'The dataset contain {rows} number of rows and {columns} number of columns.')

The dataset contain 891 number of rows and 15 number of columns.


In [21]:
df.head(3)

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True

In [22]:
df.loc[:4,['alive','alone','parch']]

alive  alone  parch
0    no  False      0
1   yes  False      0
2   yes   True      0
3   yes  False      0
4    no   True      0

In [27]:
df.loc[1:8, ['sex','fare']]

sex     fare
1  female  71.2833
2  female   7.9250
3  female  53.1000
4    male   8.0500
5    male   8.4583
6    male  51.8625
7    male  21.0750
8  female  11.1333

In [29]:
df.iloc[1:5, 2:6]

sex   age  sibsp  parch
1  female  38.0      1      0
2  female  26.0      0      0
3  female  35.0      1      0
4    male  35.0      0      0

In [30]:
df.tail(3)

survived  pclass     sex   age  sibsp  parch   fare embarked  class  \
888         0       3  female   NaN      1      2  23.45        S  Third   
889         1       1    male  26.0      0      0  30.00        C  First   
890         0       3    male  32.0      0      0   7.75        Q  Third   

       who  adult_male deck  embark_town alive  alone  
888  woman       False  NaN  Southampton    no  False  
889    man        True    C    Cherbourg   yes   True  
890    man        True  NaN   Queenstown    no   True

In [31]:
df.loc[2:7,['sex','age','alive']]

sex   age alive
2  female  26.0   yes
3  female  35.0   yes
4    male  35.0    no
5    male   NaN    no
6    male  54.0    no
7    male   2.0    no

In [32]:
df.iloc[1:5, 2:7]

sex   age  sibsp  parch     fare
1  female  38.0      1      0  71.2833
2  female  26.0      0      0   7.9250
3  female  35.0      1      0  53.1000
4    male  35.0      0      0   8.0500

In [33]:
df

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
0           0       3    male  22.0      1      0   7.2500        S   Third   
1           1       1  female  38.0      1      0  71.2833        C   First   
2           1       3  female  26.0      0      0   7.9250        S   Third   
3           1       1  female  35.0      1      0  53.1000        S   First   
4           0       3    male  35.0      0      0   8.0500        S   Third   
..        ...     ...     ...   ...    ...    ...      ...      ...     ...   
886         0       2    male  27.0      0      0  13.0000        S  Second   
887         1       1  female  19.0      0      0  30.0000        S   First   
888         0       3  female   NaN      1      2  23.4500        S   Third   
889         1       1    male  26.0      0      0  30.0000        C   First   
890         0       3    male  32.0      0      0   7.7500        Q   Third   

       who  adult_male deck  embark_town alive  alone  
0      man        True  NaN  Southampton    no  False  
1    woman       False    C    Cherbourg   yes  False  
2    woman       False  NaN  Southampton   yes   True  
3    woman       False    C  Southampton   yes  False  
4      man        True  NaN  Southampton    no   True  
..     ...         ...  ...          ...   ...    ...  
886    man        True  NaN  Southampton    no   True  
887  woman       False    B  Southampton   yes   True  
888  woman       False  NaN  Southampton    no  False  
889    man        True    C    Cherbourg   yes   True  
890    man        True  NaN   Queenstown    no   True  

[891 rows x 15 columns]

In [39]:
df.loc[0:14,['sex','age','alone']]
df.head(2)

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False

In [46]:
df.loc[(df['embark_town'] == 'Southampton') & (df['alive'] == 'yes')].count()

survived       217
pclass         217
sex            217
age            201
sibsp          217
parch          217
fare           217
embarked       217
class          217
who            217
adult_male     217
deck            80
embark_town    217
alive          217
alone          217
dtype: int64